### ISSUE TITLE
Multiple query parameters are replaced by one in 106.6

### ISSUE THREAD

**Issue by jlindner** [restsharp/RestSharp#1241]
.net Core. Using RestSharp 106.5.4, this was my request was valid. The Priorities endpoint supports a `List<string> priority` as a query parameter.

```
string server = "http://localhost:11111";
var client = new RestClient(server);
string filter = "?priority=Low&priority=Medium"
var request = new RestRequest($"priorities{filter}", Method.GET);
var response = client.Execute<Priorities>(request);
```

## Expected Behavior
When I view the response objects ResponseUri property, it has http://localhost:11111/priorities?priority=Low&priority=Medium

## Actual Behavior
With RestSharp 106.6.2, it changes my URI and I get this ResponseUri back

http://localhost:11111/priorities?priority=Low,Medium

## Steps to Reproduce the Problem

  1.  Use RestSharp 106.5.4  
Call an API with a List<string> as a query parameter similar to this.
?priority=Low&priority=Medium

It should maintain the priority=xxx for each one you query on, so in this example, your list in the endpoint should have 2 items in it.

  2.  Upgrade the project to use 106.6.2.  Call the same API again.  This time, the list only has one element in it, Low,Medium.  It no longer contains two items in the array.

## Specifications

  - Version:  106.6.2 vs 1.6.5.4
  - Platform:  Windows,  .Net Core 2.2
  - Subsystem:  .Net Core 2.2

## StackTrace
<details>

example code
```
string server = "http://localhost:11111";
var client = new RestClient(server);
string filter = "?priority=Low&priority=Medium"
var request = new RestRequest($"priorities{filter}", Method.GET);
Console.WriteLine(request.Resource);
Console.WriteLine(request.Parameters.Count.ToString());
```

With RestSharp 106.5.4, request.Resource was this
pipelines/3/feature-priorities?priority=1&priority=2
in request.Parameters, the count = 0
The service received the list of priorities correctly, having two elements in the List

With RestSharp 106.6.2, the request.Resource changed to this
pipelines/3/feature-priorities
There is one parameter now instead of the 0 the previous version had
and request.Parameters[0] is now {priority=1,2}
The service did NOT receive the priorities correctly, having one element in the List, and that element being comma seperated, instead of 2 elements like it should have had.

```

```

</details>

**Comment by alexeyzimarev** [comment]
We need a test for it, since it keeps coming back...

**Comment by alexeyzimarev** [comment]
Fixed in 106.6.5
